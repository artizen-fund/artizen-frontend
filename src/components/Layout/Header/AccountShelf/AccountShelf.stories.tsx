import AccountShelf from './'

export default {
  title: 'header/AccountShelf',
  component: AccountShelf,
  argTypes: {},
}

export const AccountShelfComponent = (props: any) => <AccountShelf {...props} />
