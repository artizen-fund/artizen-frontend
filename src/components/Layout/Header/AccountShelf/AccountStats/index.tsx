import styled from 'styled-components'
import { Icon } from '@components'
import { typography, palette } from '@theme'
import { rgba } from '@lib'

interface IAccountStats {
  stats: Array<{
    glyph: string
    unit: string
    label: string
  }>
}

const AccountStats = ({ stats }: IAccountStats) => {
  return (
    <Stats>
      {stats.map((stat, i) => (
        <Stat key={`stat-${i}`}>
          <Icon glyph={stat.glyph} outline />
          <Unit>{stat.unit}</Unit>
          <Label>{stat.label}</Label>
        </Stat>
      ))}
    </Stats>
  )
}

const Stats = styled.ul`
  grid-area: stats;

  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);

  li:nth-of-type(1),
  li:nth-of-type(3) {
    border-right: 0.5px solid ${rgba(palette.stone)};
    @media (prefers-color-scheme: dark) {
      border-right: 0.5px solid ${rgba(palette.barracuda, 0.64)};
    }
  }

  li:nth-of-type(1),
  li:nth-of-type(2) {
    border-bottom: 0.5px solid ${rgba(palette.stone)};
    @media (prefers-color-scheme: dark) {
      border-bottom: 0.5px solid ${rgba(palette.barracuda, 0.64)};
    }
  }
`

const Stat = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const Unit = styled.div`
  ${typography.title.l4}
  color: ${rgba(palette.night)};
  margin-top: 0.5em;
  @media (prefers-color-scheme: dark) {
    color: ${rgba(palette.moon)};
  }
`

const Label = styled.div`
  ${typography.label.l2}
  color: ${rgba(palette.barracuda)};
`

export default AccountStats
