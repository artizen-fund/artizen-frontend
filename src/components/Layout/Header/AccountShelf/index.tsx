import styled from 'styled-components'
import { useRouter } from 'next/router'
import { Button } from '@components'
import { breakpoint, typography, palette } from '@theme'
import { rgba, logout } from '@lib'
import { IUser } from '@types'
import AccountStats from './AccountStats'

interface IAccountShelf {
  user: IUser
  hideShelf: () => void
}

const AccountShelf = ({ user, hideShelf }: IAccountShelf) => {
  const stats = [
    {
      glyph: 'donate',
      unit: '$800',
      label: 'donated',
    },
    {
      glyph: 'palette',
      unit: '42',
      label: 'collected',
    },
    {
      glyph: 'wallet',
      unit: '$0',
      label: 'collected',
    },
    {
      glyph: 'token',
      unit: '100',
      label: '$ART',
    },
  ]

  const router = useRouter()
  const goToSettings = (section: string) => {
    router.push(`/settings#${section}`)
    hideShelf()
  }

  return (
    <Wrapper>
      <Commands>
        <Top>
          {user.firstName && <Welcome>Hi {user.firstName}</Welcome>}
          {!user.firstName && <Welcome>Welcome</Welcome>}
          <Message>Thanks for supporting the future of public goods.</Message>
          <Buttons>
            <Button onClick={() => goToSettings('profile')} stretch outline level={1} glyph="face">
              Profile
            </Button>
            <Button onClick={() => goToSettings('wallet')} stretch outline level={1} glyph="wallet">
              Wallet
            </Button>
            <Button onClick={() => goToSettings('settings')} stretch outline level={1} glyph="gear">
              Settings
            </Button>
            <Button onClick={() => goToSettings('notifications')} stretch outline level={1} glyph="bell">
              Notifications
            </Button>
          </Buttons>
        </Top>
        <div>
          <Rule />
          <Button onClick={logout} stretch level={1} glyph="power" transparent>
            Sign Out
          </Button>
        </div>
      </Commands>
      <AccountStats {...{ stats }} />
      <Illustration src="/assets/qf-square.svg" />
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto;
  grid-template-areas: 'commands' 'illustration' 'stats';
  gap: 20px;
  @media only screen and (min-width: ${breakpoint.tablet}px) {
    grid-template-rows: auto;
    grid-template-columns: repeat(3, 1fr);
    grid-template-areas: 'illustration stats commands';
    gap: 48px;
  }
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    grid-template-rows: auto;
    grid-template-columns: repeat(3, 1fr);
    grid-template-areas: 'illustration stats commands';
    gap: 64px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    gap: 5vw;
  }
  @media only screen and (min-width: 1920px) {
    gap: 96px;
  }
`

const Commands = styled.div`
  grid-area: commands;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: 50px;
`

const Top = styled.div`
  display: flex;
  flex-direction: column;
  gap: 25px;
`

const Welcome = styled.h1`
  ${typography.title.l2}
  color: ${rgba(palette.night)};
  @media (prefers-color-scheme: dark) {
    color: ${rgba(palette.moon)};
  }
`

const Message = styled.h2`
  ${typography.label.l1}
  color: ${rgba(palette.barracuda)};
`

const Buttons = styled.div`
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto;
  gap: 8px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    grid-template-rows: repeat(2, 1fr);
    grid-template-columns: repeat(2, 1fr);
  }
`

const Illustration = styled.img`
  grid-area: illustration;
  width: auto;
  max-width: 100%;
  margin: auto;
`

const Rule = styled.hr`
  appearance: none;
  width: 100%;
  height: 1px;
  background: ${rgba(palette.barracuda)};
  border: 0;
  outline: 0;
`

export default AccountShelf
