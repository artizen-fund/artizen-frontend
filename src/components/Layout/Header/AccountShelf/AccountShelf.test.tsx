import renderer from 'react-test-renderer'
import AccountShelf from './'

describe('AccountShelf', () => {
  const user: ArtizenUser = {
    id: 'abc123',
    email: 'herp@derp.com',
    issuer: 'Derp Donk',
    publicAddress: 'ICU81MI',
    phoneNumber: '555-555-1234',
  }
  it('renders correctly', () => {
    const tree = renderer.create(<AccountShelf {...{ user }} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
