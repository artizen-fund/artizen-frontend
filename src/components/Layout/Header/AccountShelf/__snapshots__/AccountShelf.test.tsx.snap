// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountShelf renders correctly 1`] = `
<div
  className="sc-a64ee2a1-0 kqTaqp"
>
  <div
    className="sc-a64ee2a1-1 hULhlL"
  >
    <h1
      className="sc-a64ee2a1-2 eXNwXt"
    >
      Welcome
    </h1>
    <h2
      className="sc-a64ee2a1-3 hzPfGl"
    >
      Thanks for supporting the future of public goods.
    </h2>
    <div
      className="sc-a64ee2a1-4 duaXJo"
    >
      <button
        className="sc-a6b3add2-2 erODIt undefined "
        onClick={[Function]}
      >
        <div
          className="sc-9eb58f28-0 jzDiDl sc-a6b3add2-0 fCwpyU"
        />
        <span>
          Profile
        </span>
      </button>
      <button
        className="sc-a6b3add2-2 erODIt undefined "
        onClick={[Function]}
      >
        <div
          className="sc-9eb58f28-0 kuWgSW sc-a6b3add2-0 fCwpyU"
        />
        <span>
          Wallet
        </span>
      </button>
      <button
        className="sc-a6b3add2-2 erODIt undefined "
        onClick={[Function]}
      >
        <div
          className="sc-9eb58f28-0 kuWgSW sc-a6b3add2-0 fCwpyU"
        />
        <span>
          Settings
        </span>
      </button>
      <button
        className="sc-a6b3add2-2 erODIt undefined "
        onClick={[Function]}
      >
        <div
          className="sc-9eb58f28-0 kuWgSW sc-a6b3add2-0 fCwpyU"
        />
        <span>
          Notifications
        </span>
      </button>
    </div>
    <hr />
    <div
      onClick={[Function]}
    >
      Sign Out
    </div>
  </div>
  <ul
    className="sc-cf63bf03-0 jnGrA-D"
  >
    <li
      className="sc-cf63bf03-1 doApLo"
    >
      <div
        className="sc-8c7bf1ae-2 jOAuUM"
      >
        <div
          className="sc-8c7bf1ae-0 kBszjY"
          color="night"
        >
          <div
            className="sc-9eb58f28-0 iglxpF"
            color="night"
          />
        </div>
      </div>
      <div
        className="sc-cf63bf03-2 dRsowg"
      >
        $800
      </div>
      <div
        className="sc-cf63bf03-3 YunQa"
      >
        donated
      </div>
    </li>
    <li
      className="sc-cf63bf03-1 doApLo"
    >
      <div
        className="sc-8c7bf1ae-2 jOAuUM"
      >
        <div
          className="sc-8c7bf1ae-0 kBszjY"
          color="night"
        >
          <div
            className="sc-9eb58f28-0 kuWgSW"
            color="night"
          />
        </div>
      </div>
      <div
        className="sc-cf63bf03-2 dRsowg"
      >
        42
      </div>
      <div
        className="sc-cf63bf03-3 YunQa"
      >
        collected
      </div>
    </li>
    <li
      className="sc-cf63bf03-1 doApLo"
    >
      <div
        className="sc-8c7bf1ae-2 jOAuUM"
      >
        <div
          className="sc-8c7bf1ae-0 kBszjY"
          color="night"
        >
          <div
            className="sc-9eb58f28-0 kuWgSW"
            color="night"
          />
        </div>
      </div>
      <div
        className="sc-cf63bf03-2 dRsowg"
      >
        $0
      </div>
      <div
        className="sc-cf63bf03-3 YunQa"
      >
        collected
      </div>
    </li>
    <li
      className="sc-cf63bf03-1 doApLo"
    >
      <div
        className="sc-8c7bf1ae-2 jOAuUM"
      >
        <div
          className="sc-8c7bf1ae-0 kBszjY"
          color="night"
        >
          <div
            className="sc-9eb58f28-0 kuWgSW"
            color="night"
          />
        </div>
      </div>
      <div
        className="sc-cf63bf03-2 dRsowg"
      >
        100
      </div>
      <div
        className="sc-cf63bf03-3 YunQa"
      >
        $ART
      </div>
    </li>
  </ul>
  <img
    className="sc-a64ee2a1-5 fZdsHB"
    src="/images/qf-square.svg"
  />
</div>
`;
